### PR TITLE
CSS Font stack cleanup and enhancements

### PR DESCRIPTION
**Pull request by silverwind**
- Update system font stack to latest from Bootstrap
- Eliminate confusing `--fonts-regular`, remove associated wildcard selector
- Set language-specifc fonts directly on root tag and elements with `lang` attribute

Everything should work as before, but now it's easier for themes to override fonts because everything is defined on `:root`.

Note: I'm pretty sure a lot of stuff in `web_src/less/_font_i18n.less` may be over-complicated/obsolete but I don't really understand enough of asian browsers to be able to touch it.

### CODE REVIEW

**Comment by silverwind** [comment]
Maybe @CL-Jeremy can review. I think the new method is much cleaner then before because it doesn't need a wildcard selector.

File: web_src/less/_base.less [hunk 1]
```diff
@@ -168,12 +168,9 @@
...
 }
 
-:root * {
-  --fonts-regular: var(--fonts-override, var(--fonts-proportional)), "Noto Sans", "Liberation Sans", sans-serif, var(--fonts-emoji);
-}
-
+/* There is a rule that sets --fonts-proportional on body in Fomantic UI */
 textarea {
-  font-family: var(--fonts-regular);
+  font-family: var(--fonts-proportional);
```
Comment (delvh) [inline web_src/less/_base.less:173]: Why not directly inside `:root` as you did in your other PR with the accent-color?
Where do we use different fonts?
I can't think of any case where we use another font…

**Comment by CL-Jeremy** [comment]
Sorry for my long absence following the series of personal issues and incidents last year. I'm still unable to contribute regularly on GitHub and I'll still have to visit my family in CN for an indefinite period some time in the next few months. In the meantime, I'll take a look at the PR I got mentioned for (thanks).

At first glance, the only pragmatic difference I find is the placement of system-ui, which at least as of last year is known to have unwanted semantics on Windows. In the other aspect of the change, I understand the ugliness of the wildcard which introduces complexity in determining cascade priorities. I'll double check if the structure works as originally intended.

File: web_src/less/_base.less (hunk 1 above)
Comment (silverwind) [reply web_src/less/_base.less:173]: `font-family` does not inherit into `textarea` or `input`. We are mostly relying on Fomantic UI to set `font-family` (and it does for `body` and `input`), but this is the one exception. See [Fiddle](https://jsfiddle.net/silverwind/bct2yuLp/).

**Comment by silverwind** [comment]
Thanks in advance. The reason I put system-ui first is that it follows the recent recommendation of [css-tricks](https://css-tricks.com/snippets/css/system-font-stack/) and Bootstrap has been doing it since around 3 years now. I assume if there are any issues, those are to be worked out by now.

One question: the new `:root[lang]` selector would cause all text including English one to be rendered with the locale-specific fonts for example:

Screen Shot 2022-10-22 at 18 59 56
Screen Shot 2022-10-22 at 19 00 42

Do asian people prefer this behaviour, or should english text be rendered in the english fonts, like previously?

**Review by delvh — APPROVED** [review]

**Comment by silverwind** [comment]
Putting in draft until I have clarity regarding `:root[lang]` selectors.

**Comment by CL-Jeremy** [comment]
> Thanks in advance. The reason I put system-ui first is that it follows the recent recommendation of [css-tricks](https://css-tricks.com/snippets/css/system-font-stack/) and Bootstrap has been doing it since around 3 years now. I assume if there are any issues, those are to be worked out by now.

That was the status quo, and seems still is as of now. Bootstrap developers couldn't do anything about that - not without some JavaScript magic - only browser vendors could make changes to their actual interpretations. See below for more.

> 
> 
> One question: the new `:root[lang]` selector would cause all text including English one to be rendered with the locale-specific fonts for example:
> 
> 
> 
> Screen Shot 2022-10-22 at 18 59 56
> 
> Screen Shot 2022-10-22 at 19 00 42
> 
> 
> 
> Do asian people prefer this behaviour, or should english text be rendered in the english fonts, like previously?

Some Asian people could put up with it, others couldn't. In general, that's not how things suppose to work design-wise.

A bit of history: On older Windows versions, default Western web font was Arial (except ASP/ASP.NET) although the UI font was Microsoft Sans UI/Tahoma, while Simplified Chinese, for example, used SimSun extensively on both sides - a mono-/duospaced serif font that definitely did not match the look of sans-serif fonts, yet was still the resolution for sans-serif (whereas Japanese would resolve to MS Gothic). During this time, it may be safe to conclude that each language just use their own font design with no coherence at all. In fact, Chrome and previously WebKit was criticized for just picking SimSun with pixelated bitmap fonts for sans-serif even after the quasi-universal availability of Microsoft YaHei, even with pure English text, on Simplified Chinese locale, while Microsoft's IE 9 and above at least makes an effort to render them in Arial or Times New Roman with beautiful ClearType (I'm mentioning this to support my argument not to trust the platform just because it's the correct way in principle).

Starting with the release of Vista, Western UI/native web designers have increasingly become fond of the more native-looking Segoe UI on Windows, which is the branding font to promote Microsoft's ClearType technology and led a complete UI overhaul for East Asian system fonts. The Chinese (Simplified, Traditional) and Korean counterparts released along Vista feature Segoe UI with slightly tweaked metrics for Western characters, while for Japanese Microsoft shipped Meiryo/Meiryo UI with a much enlarged Tahoma design for Western glyphs. At this point, either approach (Fallback/independent) is acceptable and rightfully so: the vertical metrics from Microsoft's font match just fine, and either approach looks native enough on websites.

With the introduction of more weights, however, the separation of Western fonts from localized fonts now has a much more significant implication on design. In general, it's sort of agreed that a better differentiation of weights for Western fonts outweighs the perfect match with fallback fonts - a rule of thumb most probably originated on mobile platforms. Microsoft went on to include Yu Gothic for Japanese with Windows 8.1 along with a UI-specific version with narrow kanas - looking out of place just by itself - plus the fact that Yu Gothic's heaviest weight, Medium, is thinner than Segoe UI Regular! Hence the special workaround I have included in Gitea. In short, on the Windows landscape, the design of native-like font stacks is especially complex without a designated best practice.

How about macOS/iOS? The situation is mostly simple: the system (Core Text) is intelligent as of lately and pairs the fallback fonts to -apple-system (historical alias: BlinkMacSystemFont), so that's safe to be put at first place in the font stack. Granted, on Japanese locale, for actual system UI elements, macOS does use a downscaled version of Hiragino Sans (aka. Hiragino Kaku Gothic ProN), but just as with Yu Gothic UI, that font is never to be used for regular web design (and unsupported by some layout engines). As for even earlier systems, Mac OS X 10.6 involved the use the Lucida Grande + Aqua Kana + Hiragino Kaku Gothic Pro trilogy, but thankfully that's irrelevant now.

Lastly, Linux and other FOSS desktops. The user/admin has so much freedom to configure the UI and even the font resolution (replacements, bans and more) of the entire system based on Fontconfig, and the inclusion of license-incompatible proprietary fonts is either based on user choice or totally out of question.

Hopefully I was able to make my arrangement of the proportional font stack clear to you (again) with these analyses and anecdotes.

**Comment by silverwind** [comment]
Not going to further touch this, this font i18n stuff is black magic to me 😉

**Pull request closed without merging**